0.data_analysis/plate_correlation_analyses/nbconverted/well_agg_plate_genotype_correlation_analysis.py

```diff
@@ -0,0 +1,170 @@
+
+# Path to correlation class
+sys.path.append(
+    "../utils"
+)
+
+# Class for calculating correlations
+from CorrelateData import CorrelateData
+
+# ## Find the root of the git repo on the host system
+
+# In[2]:
+
+
+# Get the current working directory
+cwd = pathlib.Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# # Inputs
+
+# In[3]:
+
+
+data_path = pathlib.Path(root_dir / "nf1_painting_repo/3.processing_features/data/single_cell_profiles").resolve(strict=True)
+
+plate3df_path = pathlib.Path(root_dir / data_path / "Plate_3_bulk_camerons_method.parquet").resolve(strict=True)
+plate3pdf_path = pathlib.Path(root_dir / data_path / "Plate_3_prime_bulk_camerons_method.parquet").resolve(strict=True)
+plate5df_path = pathlib.Path(root_dir / data_path / "Plate_5_bulk_camerons_method.parquet").resolve(strict=True)
+
+plate3df = pd.read_parquet(plate3df_path)
+plate3pdf = pd.read_parquet(plate3pdf_path)
+plate5df = pd.read_parquet(plate5df_path)
+
+
+# # Outputs
+
+# In[4]:
+
+
+plate_correlation_path = pathlib.Path("construct_correlation_data/well_agg_plate_genotype_correlations.parquet")
+plate_correlation_path.mkdir(parents=True, exist_ok=True)
+
+
+# # Process Bulk Plate Data
+
+# ## Combine data
+# Concat plate data and retain common columns.
+
+# In[5]:
+
+
+plates_cols = plate3df.columns.intersection(plate3pdf.columns).intersection(plate5df.columns)
+platesdf = pd.concat([plate3df, plate3pdf, plate5df], axis=0)
+platesdf = platesdf[plates_cols]
+
+
+# In[6]:
+
+
+# Morphology and metadata columns
+morph_cols = [col for col in platesdf.columns if "Metadata" not in col]
+meta_cols = platesdf.columns.difference(morph_cols)
+
+
+# # Correlate wells
+# Wells are correlated between plate and genotype.
+
+# In[7]:
+
+
+cd = CorrelateData()
+correlationsdf = []
+
+
+# ## Well Correlations (same genotypes different plates)
+
+# In[8]:
+
+
+for genotype in platesdf["Metadata_genotype"].unique():
+
+    correlation_params = {
+    }
+
+    correlationsdf.append(
+        cd.inter_correlations(
+            _df=platesdf.loc[platesdf["Metadata_genotype"] == genotype].copy(),
+            _antehoc_group_cols=["Metadata_Plate"],
+            _feat_cols=morph_cols,
+            _posthoc_group_cols=["Metadata_Well", "Metadata_genotype"],
+            _drop_cols=["Metadata_Well"]
+        )
+    )
+
+
+# ## Well Correlations (different genotypes and all possible plates)
+
+# In[9]:
+
+
+correlationsdf.append(
+    cd.inter_correlations(
+        _df=platesdf.copy(),
+        _antehoc_group_cols=["Metadata_genotype"],
+        _feat_cols=morph_cols,
+        _posthoc_group_cols=["Metadata_Plate", "Metadata_Well"],
+        _drop_cols=["Metadata_Well"]
+    )
+)
```

Comment:
Just curious, do these correlations need to be done in different functions? Would it not be possible to run this function once with all of the data and get the same output?

Reply:
I think this was the implementation from a previous correlation analysis. However, I wanted to modularize and broaden the scope where this code can be applied, such as with siRNA treatment data. There is a proposed enhancement (see #59) to abstract away the lower level details, however, for now I think this is a lower priority. I think modularizing the code improved readability, especially with the comments both in this notebook and the correlation util.